crates/snapbox/Cargo.toml

```diff
@@ -48,6 +48,10 @@ detect-encoding = ["content_inspector"]
 path = ["tempfile", "walkdir", "dunce", "detect-encoding", "filetime"]
 ## Snapshotting of commands
 cmd = ["os_pipe", "wait-timeout"]
+## Snapshotting of json
+json = ["serde_json", "detect-extension"]
+
+detect-extension = []
```

Comment:
`detect-encoding` is mostly to control the presence of a dependency.  Not sure if a `detect-extension` is worth it (or maybe we bundle it with `detect-encoding`?)

Reply:
I can make whatever work. I just created an empty one so that the different formats would turn on extension detection independently